src/transformers/models/opt/modeling_opt.py

```diff
@@ -93,30 +114,30 @@ class OPTAttention(nn.Module):
 
     def __init__(
         self,
-        embed_dim: int,
-        num_heads: int,
-        dropout: float = 0.0,
-        is_decoder: bool = False,
-        bias: bool = True,
+        config: OPTConfig,
```

Comment:
Are we ok modifying the Attention Class's signature? We've had issues in the past when changing signature of such classes (e.g. for Llama when we added `padding_mask`) cc @amyeroberts

Reply:
It should be OK, as this isn't a class importable from the top layer of transformers or in the documentation. However, it definitely can still cause issues! What I'd suggest is something like this (happy for you to modify as desired): 

```py
    def __init__(
        self,
        config,
        is_decoder: bool = False,
        **kwargs
    ):
	def _handle_deprecated_argument(config_arg_name, config, fn_arg_name, kwargs):
	    """
	    If a the deprecated argument `fn_arg_name` is passed, raise a deprecation 
	    warning and return that value, otherwise take the equivalent config.config_arg_name
	    """
	    val = None
	    if fn_arg_name in kwargs:
	    	logging.warning(
	       		"Passing in {} to {self.__class__.__name__} is deprecated and won't be supported from v4.38."
	       		" Please set it in the config instead"
	       	)
	       	val = kwargs.pop(fn_arg_name)
	    else:
	       	val = getattr(config, config_arg_name)
            return val

        embed_dim = _handle_deprecated_argument("hidden_size", config, "embed_dim", kwargs)
        num_heads = _handle_deprecated_argument("num_attention_heads", config, "num_heads", kwargs)
        dropout = _handle_deprecated_argument("attention_dropout", config, "dropout", kwargs)
        enable_bias = _handle_deprecated_argument("enable_bias", config, "bias", kwargs)
```